flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/FsCheckpointStorage.java

```diff
@@ -107,6 +105,13 @@ public CheckpointStorageLocation initializeLocationForCheckpoint(long checkpoint
 		// prepare all the paths needed for the checkpoints
 		final Path checkpointDir = createCheckpointDirectory(checkpointsDirectory, checkpointId);
 
+		if (!areDirectoriesCreated) {
+			fileSystem.mkdirs(checkpointsDirectory);
```

Comment:
If you want to minimize calls, this one should not even be needed because it would simply happen as part of the `fileSystem.mkdirs(checkpointDir);` below.